src/datacollector/DataCollectionAttachmentManager.cs

```diff
@@ -196,86 +151,47 @@ public void AddAttachment(FileTransferInformationExtension fileTransferInfo)
-            this.AttachmentRequests.Clear();
             this.SessionOutputDirectory = null;
-
-            this.Dispose(true);
-
-            // Use SupressFinalize in case a subclass
-            // of this type implements a finalizer.
-            GC.SuppressFinalize(this);
+            this.AttachmentSets.Clear();
+            this.attachmentTasks.Clear();
         }
 
         #endregion
 
-        /// <summary>
-        /// The dispose.
-        /// </summary>
-        /// <param name="disposing">
-        /// The disposing.
-        /// </param>
-        protected virtual void Dispose(bool disposing)
-        {
-            if (!this.disposed)
-            {
-                if (disposing)
-                {
-                    this.fileCopierJobQueue?.Dispose();
-                }
-
-                this.disposed = true;
-            }
-        }
-
         #region private methods
 
         /// <summary>
-        /// Mark the request as processed/completed. 
-        /// If any error occurred during processing, error information is also sent to copy request.
-        /// </summary>
-        /// <param name="attachmentRequest">
-        /// The file copy request.
-        /// </param>
-        /// <param name="e">
-        /// The e.
-        /// </param>
-        private static void CompleteFileTransfer(AttachmentRequest attachmentRequest, Exception e)
-        {
-            attachmentRequest.CompleteRequest(e);
-        }
-
-        /// <summary>
         /// Sanity checks on CopyRequestData 
         /// </summary>
-        /// <param name="attachmentRequest">
-        /// The copy Request.
+        /// <param name="fileTransferInfo">
+        /// The file Transfer Info.
         /// </param>
-        private static void Validate(AttachmentRequest attachmentRequest)
+        /// <param name="localFilePath">
+        /// The local File Path.
+        /// </param>
+        private static void Validate(FileTransferInformation fileTransferInfo, string localFilePath)
         {
-            if (!File.Exists(attachmentRequest.FileTransferInfo.FileName))
+            if (!File.Exists(fileTransferInfo.FileName))
             {
                 throw new FileNotFoundException(
                     string.Format(
                         CultureInfo.CurrentCulture,
                         "Could not find source file '{0}'.",
-                        attachmentRequest.FileTransferInfo.FileName));
+                        fileTransferInfo.FileName));
             }
 
-            var directoryName = Path.GetDirectoryName(attachmentRequest.LocalFilePath);
+            var directoryName = Path.GetDirectoryName(localFilePath);
+
             if (!Directory.Exists(directoryName))
             {
                 Directory.CreateDirectory(directoryName);
             }
-            else if (File.Exists(attachmentRequest.LocalFilePath))
+            else if (File.Exists(localFilePath))
```

Comment:
This should be logged that we're removing existing file. What happens if we cannot remove the file?